app/Http/Controllers/EntryController.php

```diff
@@ -97,11 +100,20 @@ public function store(EntryStoreRequest $request)
             CreateAndAttachRating::run($entry, $ratingData);
         }
 
-        if ($request->type == Entry::TYPE_DM) {
-            return redirect()->route('dm-entry.index');
+        if (isset($campaign)) {
+            $redirectRoute = 'campaign.show';
+            $parameter = $campaign;
+        } elseif (!isset($campaign)) {
+            if ($entryData['type'] === Entry::TYPE_DM) {
+                $redirectRoute = 'dm-entry.index';
+                $parameter = null;
+            } else {
+                $redirectRoute = 'character.show';
+                $parameter = $entry->character_id;
+            }
```

Comment:
Typically i'd avoid nesting control flow logic more than 1 layer deep if possible. Since we don't really have the luxary of time, ill just say this is a refactor target and move on